visvis2/geometry/_base.py

```diff
@@ -1,5 +1,9 @@
+import numpy as np
+
+
 class Geometry:
-    def __init__(self):
-        self.index = None
-        self.vertex_data = []
+    def __init__(self, index=None, *vertex_data):
+        # todo: don't cast if an array is already given
+        self.index = None if index is None else np.array(index, np.uint32)
+        self.vertex_data = [np.array(array, np.float32) for array in vertex_data]
```

Comment:
why is this a list instead of a dict? don't you need to know if something is for `position` or `color` for example?

Reply:
Because binding in the shader is based on integers. Unless we put some sauce on top to bind with strings ...

Reply:
:tomato: